plugins/processors/awsentity/processor.go

```diff
@@ -0,0 +1,413 @@
+// SPDX-License-Identifier: MIT
+
+package awsentity
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	semconv "go.opentelemetry.io/collector/semconv/v1.22.0"
+	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/extension/entitystore"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsentity/entityattributes"
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsentity/internal/k8sattributescraper"
+	"github.com/aws/amazon-cloudwatch-agent/translator/config"
+)
+
+const (
+	attributeAwsLogGroupNames      = "aws.log.group.names"
+	attributeDeploymentEnvironment = "deployment.environment"
+	attributeServiceName           = "service.name"
+	attributeService               = "Service"
+	EMPTY                          = ""
+)
+
+type scraper interface {
+	Scrape(rm pcommon.Resource)
+	Reset()
+}
+
+type EC2ServiceAttributes struct {
+	InstanceId        string `validate:"required"`
+	AutoScalingGroup  string `validate:"omitempty"`
+	ServiceNameSource string `validate:"omitempty"`
+}
+
+type K8sServiceAttributes struct {
+	Cluster           string `validate:"required"`
+	Namespace         string `validate:"required"`
+	Workload          string `validate:"required"`
+	Node              string `validate:"required"`
+	InstanceId        string `validate:"omitempty"`
+	ServiceNameSource string `validate:"omitempty"`
+}
+
+// use a single instance of Validate, it caches struct info
+var validate = validator.New(validator.WithRequiredStructEnabled())
+
+// exposed as a variable for unit testing
+var addToEntityStore = func(logGroupName entitystore.LogGroupName, serviceName string, environmentName string) {
+	es := entitystore.GetEntityStore()
+	if es == nil {
+		return
+	}
+	es.AddServiceAttrEntryForLogGroup(logGroupName, serviceName, environmentName)
+}
+
+var addPodToServiceEnvironmentMap = func(podName string, serviceName string, environmentName string, serviceNameSource string) {
+	es := entitystore.GetEntityStore()
+	if es == nil {
+		return
+	}
+	es.AddPodServiceEnvironmentMapping(podName, serviceName, environmentName, serviceNameSource)
+}
+
+var getEC2InfoFromEntityStore = func() entitystore.EC2Info {
+	es := entitystore.GetEntityStore()
+	if es == nil {
+		return entitystore.EC2Info{}
+	}
+
+	return es.EC2Info()
+}
+
+var getServiceNameSource = func() (string, string) {
+	es := entitystore.GetEntityStore()
+	if es == nil {
+		return EMPTY, EMPTY
+	}
+	return es.GetMetricServiceNameAndSource()
+}
+
+// awsEntityProcessor looks for metrics that have the aws.log.group.names and either the service.name or
+// deployment.environment resource attributes set, then adds the association between the log group(s) and the
+// service/environment names to the entitystore extension.
+type awsEntityProcessor struct {
+	config     *Config
+	k8sscraper scraper
```

Comment:
nit: Why are we using an interface? It doesn't look like it serves a purpose and it forces us to type assert to get the `k8sattributescraper` struct.

Reply:
Non-blocking comment